watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/metab/metab.py

```diff
@@ -0,0 +1,628 @@
+            ) / m.fs.costing.annual_hydrogen_production
+        elif c == "reactor":
+            return (
+                m.fs.metab_hydrogen.costing.DCC_reactor
+                * m.fs.costing.TIC
+                * (
+                    m.fs.costing.capital_recovery_factor
+                    + m.fs.costing.maintenance_costs_percent_FCI
+                )
+            ) / m.fs.costing.annual_hydrogen_production
+        elif c == "mixer":
+            return (
+                m.fs.metab_hydrogen.costing.DCC_mixer
+                * m.fs.costing.TIC
+                * (
+                    m.fs.costing.capital_recovery_factor
+                    + m.fs.costing.maintenance_costs_percent_FCI
+                )
+            ) / m.fs.costing.annual_hydrogen_production
+        elif c == "vacuum":
+            return (
+                m.fs.metab_hydrogen.costing.DCC_vacuum
+                * m.fs.costing.TIC
+                * (
+                    m.fs.costing.capital_recovery_factor
+                    + m.fs.costing.maintenance_costs_percent_FCI
+                )
+            ) / m.fs.costing.annual_hydrogen_production
+        elif c == "membrane":
+            return (
+                m.fs.metab_hydrogen.costing.DCC_membrane
+                * m.fs.costing.TIC
+                * (
+                    m.fs.costing.capital_recovery_factor
+                    + m.fs.costing.maintenance_costs_percent_FCI
+                )
+            ) / m.fs.costing.annual_hydrogen_production
+        elif c == "electricity_vacuum":
+            return (
+                pyunits.convert(
+                    m.fs.metab_hydrogen.energy_electric_vacuum_flow_vol_byproduct
+                    * m.fs.metab_hydrogen.properties_byproduct[0].flow_mass_comp[
+                        "hydrogen"
+                    ]
+                    * m.fs.costing.electricity_cost,
+                    to_units=m.fs.costing.base_currency / m.fs.costing.base_period,
+                )
+                * m.fs.costing.utilization_factor
+            ) / m.fs.costing.annual_hydrogen_production
+        elif c == "electricity_mixer":
+            return (
+                pyunits.convert(
+                    m.fs.metab_hydrogen.energy_electric_mixer_vol
+                    * m.fs.metab_hydrogen.volume
+                    * m.fs.costing.electricity_cost,
+                    to_units=m.fs.costing.base_currency / m.fs.costing.base_period,
+                )
+                * m.fs.costing.utilization_factor
+            ) / m.fs.costing.annual_hydrogen_production
+        elif c == "heat":
+            return (
+                pyunits.convert(
+                    m.fs.metab_hydrogen.heat[0] * m.fs.costing.heat_cost,
+                    to_units=m.fs.costing.base_currency / m.fs.costing.base_period,
+                )
+                * m.fs.costing.utilization_factor
+            ) / m.fs.costing.annual_hydrogen_production
+        else:
+            return 0
```

Comment:
I'm not really a fan of this, or the similar "rule" below. This is an instance where *not* using a rule seems clearer. E.g.

```python
m.fs.costing.LC_comp = Set(
    initialize=[
            "bead",
            "reactor",
            "mixer",
            "membrane",
            "vacuum",
            "heat",
            "electricity_mixer",
            "electricity_vacuum",
            "hydrogen_product",
            "methane_product",
    ]
)
m.fs.costing.LCOH_comp = Expression(m.fs.costing.LC_comp)

m.fs.costing.LCOH_comp["bead"] =  (
                m.fs.metab_hydrogen.costing.DCC_bead
                * m.fs.costing.TIC
                * (
                    m.fs.costing.capital_recovery_factor
                    + m.fs.costing.maintenance_costs_percent_FCI
                )
                + m.fs.metab_hydrogen.costing.fixed_operating_cost
            ) / m.fs.costing.annual_hydrogen_production
            
 m.fs.costing.LCOH_comp["reactor"] = (
                m.fs.metab_hydrogen.costing.DCC_reactor
                * m.fs.costing.TIC
                * (
                    m.fs.costing.capital_recovery_factor
                    + m.fs.costing.maintenance_costs_percent_FCI
                )
            ) / m.fs.costing.annual_hydrogen_production
 
```
etc. If certain indices *really* aren't needed then the index set should probably be customized.

Reply:
Right- so this was something that was thrown together for quickly churning out an analysis. I agree that it could be made clearer and cleaner. @TimBartholomew feel free to jump in here with any thoughts about how this was formulated.

I made an issue (#507 ) to revisit how this was done essentially because the sum of these components do not exactly match the total value from the corresponding expression(s). 

It's worth noting that this use case depended on some initial parameter estimates that resulted in brow-raising results, and we intend to reiterate and refine this whole thing.